src/EGraphs/saturation.jl

```diff
@@ -115,40 +115,97 @@ function eqsat_search!(
   return n_matches
 end
 
-function instantiate_enode!(bindings, @nospecialize(g::EGraph), p::PatLiteral)::Id
+function instantiate_enode!(bindings::Bindings, isliteral_bitvec::UInt64, g::EGraph, p::PatLiteral)::Id
   add_constant_hashed!(g, p.value, v_head(p.n))
   add!(g, p.n, true)
 end
 
-instantiate_enode!(bindings, @nospecialize(g::EGraph), p::PatVar)::Id = v_pair_first(bindings[p.idx])
-function instantiate_enode!(bindings, g::EGraph{ExpressionType}, p::PatExpr)::Id where {ExpressionType}
+function instantiate_enode!(bindings::Bindings, isliteral_bitvec::UInt64, g::EGraph, p::PatVar)::Id
+  # @show p
+  # @show isliteral_bitvec
+  # @show bindings
+  if v_bitvec_check(isliteral_bitvec, p.idx)
+    add!(g, VecExpr(Id[0, 0, 0, bindings[p.idx]]), true)
```

Comment:
This line seems odd. Should be double-checked.

Reply:
It's not the best solution, but if a pattern variable is bound to a literal which is present in the e-graph, then it's re-adding the literal e-node (thus akin to a lookup). I'm thinking that we probably don't need a full VecExpr for literals but previous experiments showed a downgrade in performance when using just hashes stored separately for literals.

There is likely a better solution for retrieving an e-class id from a literal hash